### PR TITLE
feat(indexer): add annoy and nmslib

### PR DESCRIPTION
**Pull request by kashyapvad**

### CODE REVIEW

**Comment by nan-wang** [comment]
hi, @kashyapvad Installing faiss needs not only `pip install` but also compilation the c++ codes. For more details, please refer to [faiss INSTALLATION](https://github.com/facebookresearch/faiss/blob/master/INSTALL.md#general-compilation-instructions). We are working on building a docker image for faiss. Anyway, thanks for your contribution.

**Comment by kashyapvad** [comment]
Hi @nan-wang. Thank you for your reply and clarification. I've also implemented annoy and nmslib images. Should I remove faiss and commit just the two of them for now?

File: hub/executors/indexers/vector/annoy/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM jinaai/jina:devel
+
+RUN apt-get update && apt-get install --no-install-recommends -y g++
+RUN pip install annoy
+
+RUN pip install jina
+
+ENV TMP_WORKSPACE /tmp/jina/annoy/
```
Comment (nan-wang) [inline hub/executors/indexers/vector/annoy/Dockerfile:8]:
```suggestion

```

File: hub/executors/indexers/vector/annoy/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+FROM jinaai/jina:devel
+
+RUN apt-get update && apt-get install --no-install-recommends -y g++
+RUN pip install annoy
+
+RUN pip install jina
```
Comment (nan-wang) [inline hub/executors/indexers/vector/annoy/Dockerfile:6]: `jina` is already included in the base image.
```suggestion
```

File: hub/executors/indexers/vector/annoy/annoy.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+!AnnoyIndexer
+metas:
+  workspace: $TMP_WORKSPACE
```
Comment (nan-wang) [inline hub/executors/indexers/vector/annoy/annoy.yml:3]: For indexer, it's better to use the `--volumes` to amount the volume. Here is an example in the faiss indexer. [https://github.com/jina-ai/jina-hub/tree/master/hub/executors/indexers/vector/faiss#usages](https://github.com/jina-ai/jina-hub/tree/master/hub/executors/indexers/vector/faiss#usages)
```suggestion
  workspace: ./workspace
```

File: hub/executors/indexers/vector/annoy/annoy.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+  name: indexer_annoy
+  py_modules: annoy.py
```
Comment (nan-wang) [inline hub/executors/indexers/vector/annoy/annoy.yml:5]: `AnnoyIndexer` is a built-in class, and therefore you can use it out-of-box. `py_modules` is useful when you want to implement your own executors.
```suggestion
```

File: hub/executors/indexers/vector/nmslib/nmslib.py [hunk 5]
```diff
@@ -0,0 +1 @@
+from jina.executors.indexers.vector.nmslib import NmslibIndexer
```
Comment (nan-wang) [inline hub/executors/indexers/vector/nmslib/nmslib.py:1]:
```suggestion
```

File: hub/executors/indexers/vector/nmslib/manifest.yml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+name: nmslib
+description: an image with the indexer nmslib
+author: Jina AI Dev-Team (dev-team@jina.ai)
```
Comment (nan-wang) [inline hub/executors/indexers/vector/nmslib/manifest.yml:3]: We'd love to give credits to you! Please have your name typed here. ;)

File: hub/executors/indexers/vector/nmslib/nmslib.yml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+!NmslibIndexer
+metas:
+  workspace: $TMP_WORKSPACE
+  name: indexer_nmslib
+  py_modules: nmslib.py
```
Comment (nan-wang) [inline hub/executors/indexers/vector/nmslib/nmslib.yml:5]:
```suggestion
```

File: hub/executors/indexers/vector/nmslib/nmslib.yml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+!NmslibIndexer
+metas:
+  workspace: $TMP_WORKSPACE
```
Comment (nan-wang) [inline hub/executors/indexers/vector/nmslib/nmslib.yml:3]:
```suggestion
  workspace: ./workspace
```

File: hub/executors/indexers/vector/nmslib/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,10 @@
+FROM jinaai/jina:devel
+
+RUN pip install nmslib
+
+RUN pip install jina
```
Comment (nan-wang) [inline hub/executors/indexers/vector/nmslib/Dockerfile:5]:
```suggestion
```

File: hub/executors/indexers/vector/annoy/manifest.yml [hunk 10]
```diff
@@ -0,0 +1,9 @@
+name: annoy
+description: an image with the indexer annoy
+author: Jina AI Dev-Team (dev-team@jina.ai)
```
Comment (nan-wang) [inline hub/executors/indexers/vector/annoy/manifest.yml:3]: We'd love to give credits to you! Please have your name typed here. ;)
```suggestion
author: Jina AI Dev-Team (dev-team@jina.ai)
```

File: hub/executors/indexers/vector/annoy/annoy.py [hunk 11]
```diff
@@ -0,0 +1 @@
+from jina.executors.indexers.vector.annoy import AnnoyIndexer
```
Comment (nan-wang) [inline hub/executors/indexers/vector/annoy/annoy.py:1]:
```suggestion
```

File: hub/executors/indexers/vector/nmslib/Dockerfile [hunk 12]
```diff
@@ -0,0 +1,10 @@
...
+
+ENV TMP_WORKSPACE /tmp/jina/nmslib/
```
Comment (nan-wang) [inline hub/executors/indexers/vector/nmslib/Dockerfile:7]:
```suggestion
```

File: hub/executors/indexers/vector/annoy/Dockerfile (hunk 2 above)
Comment (kashyapvad) [reply hub/executors/indexers/vector/annoy/Dockerfile:6]: hadn't realized I'm installing the image again. removed it.

**Comment by kashyapvad** [comment]
Made the changes requested

**Pull request closed without merging**